### PR TITLE
feat: add auth support for downloading OCI artifacts

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
Add authentication support for downloading OCI artifacts such as Trivy DB.

```
$ TRIVY_USERNAME=foo TRIVY_PASSWORD=bar trivy image --db-repository ghcr.io/your/private-db --download-db-only
```

Trivy currently supports keychains for private OCI artifacts. It requires `docker login` or something similar to configure credentials for registries. This PR allows taking credentials from environmental variables, CLI flags and `trivy.yaml`. It was implemented for private container images so that Trivy could scan them for vulnerabilities. This PR expands it to OCI artifacts as well.

Note that Java DB is out of scope in this PR.

- [x] Trivy DB (default: ghcr.io/aquasecurity/trivy-db)
- [x] Misconfiguration policies
- [x] Wasm modules
- [ ] Java DB

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/3891

## Related PRs
- [x] https://github.com/aquasecurity/trivy/pull/3906

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Review by DmitriyLewen — COMMENTED** [review]
Good work!
I added 1 comment.

Also I think we need to add information about support of private DB repositories and combination of credentials for DB and images.

File: pkg/commands/artifact/run.go [hunk 1]
```diff
@@ -73,6 +73,9 @@ type ScannerConfig struct {
 
 	// Artifact options
 	ArtifactOption artifact.Option
+
+	// Registry options
+	RemoteOption ftypes.RemoteOptions
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/commands/artifact/run.go]: I've removed it in this commit.
https://github.com/aquasecurity/trivy/pull/3906/commits/554e01c19f380c2374288cdfca138a80b6546ede

I probably made a mistake when rebasing, and It has reappeared. Removed it again.
https://github.com/aquasecurity/trivy/pull/3915/commits/e2630d03abfdd2d136105befa252a611896e6060

**Comment by knqyf263** [comment]
I updated the documentation. I'll restructure the entire doc shortly, though.
https://github.com/aquasecurity/trivy/pull/3915/commits/74a1aa4998d8e3c26482fb81c4365f20d5acf5ff

**Review by DmitriyLewen — APPROVED** [review]
LGTM

**Comment by chen-keinan** [comment]
@knqyf263 this change is not supported for `trivy server ...` sub command ? 

```shell
TRIVY_USERNAME=user TRIVY_PASSWORD=pass trivy server --listen 0.0.0.0:4975 --db-repository ghcr.io/private-repo/trivy-db  --download-db-only
```

**Comment by knqyf263** [comment]
Right, server is not supported.

**Comment by chen-keinan** [comment]
> Right, server is not supported now.

Any plan to add support to server  , the initial use-case in `trivy-operator` was for server subcommand

**Comment by knqyf263** [comment]
Yes, it is just adding 1 line here, like `RegistryFlagGroup: flag.NewRegistryFlags(),`
https://github.com/aquasecurity/trivy/blob/f14bed453246a880854ff2ee8cdd1a483e4ebc4c/pkg/commands/app.go#L520-L525

But if you just want to download the database with `--download-db-only`, you can use other subcommands until `server` supports it.

```
TRIVY_USERNAME=user TRIVY_PASSWORD=pass trivy image --db-repository ghcr.io/private-repo/trivy-db  --download-db-only
```

Anyway, it is good to add registry flags to `server`. @chen-keinan Could you work on this change and test it with `trivy-operator`?

**Pull request merged**